src/datumaro/plugins/data_formats/cityscapes.py

```diff
@@ -299,49 +308,64 @@ def _load_items(self):
                 recursive=True,
             )
             mask_suffix = CityscapesPath.GT_INSTANCE_MASK_SUFFIX
+
+        self._image_mask_path_by_id = {}
         for mask_path in masks:
             item_id = self._get_id_from_mask_path(mask_path, mask_suffix)
 
-            anns = []
-            instances_mask = lazy_image(mask_path, dtype=np.int32)
-            segm_ids = np.unique(instances_mask())
-            for segm_id in segm_ids:
-                # either is_crowd or ann_id should be set
-                if segm_id < 1000:
-                    label_id = segm_id
-                    is_crowd = True
-                    ann_id = None
-                else:
-                    label_id = segm_id // 1000
-                    is_crowd = False
-                    ann_id = segm_id % 1000
-                anns.append(
-                    ExtractedMask(
-                        index_mask=instances_mask,
-                        index=segm_id,
-                        label=label_id,
-                        id=ann_id,
-                        attributes={"is_crowd": is_crowd},
-                    )
-                )
-
             image = image_path_by_id.pop(item_id, None)
-            if image:
-                image = Image.from_file(path=image)
-
-            items[item_id] = DatasetItem(
-                id=item_id, subset=self._subset, media=image, annotations=anns
-            )
+            self._image_mask_path_by_id[item_id] = (image, mask_path)
 
         for item_id, path in image_path_by_id.items():
-            items[item_id] = DatasetItem(
-                id=item_id, subset=self._subset, media=Image.from_file(path=path)
-            )
+            self._image_mask_path_by_id[item_id] = (path, None)
 
         self._categories = self._load_categories(
             self._path, use_train_label_map=mask_suffix is CityscapesPath.LABEL_TRAIN_IDS_SUFFIX
         )
-        return items
+
+    def _mask_path_to_annotations(self, mask_path: Optional[str]) -> list[Annotation]:
+        anns = []
+        if not mask_path:
+            return anns
+        instances_mask = lazy_image(mask_path, dtype=np.int32)
+        segm_ids = np.unique(instances_mask())
+        for segm_id in segm_ids:
+            # either is_crowd or ann_id should be set
+            if segm_id < 1000:
+                label_id = segm_id
+                is_crowd = True
+                ann_id = None
+            else:
+                label_id = segm_id // 1000
+                is_crowd = False
+                ann_id = segm_id % 1000
+            anns.append(
+                ExtractedMask(
+                    index_mask=instances_mask,
+                    index=segm_id,
+                    label=label_id,
+                    id=ann_id,
+                    attributes={"is_crowd": is_crowd},
+                )
+            )
+            self._ann_types.add(AnnotationType.mask)
```

Comment:
I think this should happen if we find at least 1 mask path. There are no other possible annotation types in this format, and there is no need parse masks to conclude that there are any masks in the dataset (all masks must contain annotations).